sos/archive.py

```diff
@@ -209,7 +210,14 @@ def add_node(self, path, mode, device):
         dest = self.dest_path(path)
         self._check_path(dest)
         if not os.path.exists(dest):
-            os.mknod(dest, mode, device)
+            try:
+                os.mknod(dest, mode, device)
+            except OSError as e:
```

Comment:
The traditional way of dealing with this in cases where you are root but not allowed to mknod is to bind-mount the source path to the target rather than copy. So in this case, touch dest and bind-mount path onto dest.